code-dot-org/code-dot-org
# update .script_json and scripts.en.yml on lesson save

Finishes [PLAT-564](https://codedotorg.atlassian.net/browse/PLAT-564). When the lesson edit page is saved:
1. scripts.en.yml is updated with the new lesson name. this makes it so that the script overview page and teacher-facing lesson plan show the correct lesson name.
2. update the .script_json file for the lesson's script. this has no other visible result because we're not seeding the result back into the database yet.

I'm not updating the .script file on lesson save. the reason is that for migrated scripts I'm expecting that the only part of the .script file we'll look at is the part that says `is_migrated: true`, which won't change during lesson edit. 

## Testing story

unit tests
* verify output of i18n_hash
* verify that both files are updated in levelbuilder mode
* verify that neither file is updated without levelbuilder mode (for example, in the test environment)

manual testing
1. save the script edit page for a migrated script, ensuring that .script_json and scripts.en.yml are up to date
2. navigate to the lesson edit page
3. rename the lesson, and rename an activity
4. save the page
5. verified that scripts.en.yml shows the new lesson name, with no other changes
6. verified that .script_json shows the new lesson name and new lesson_activity name, with no other changes
7. verify that the lesson and activity names now appear correctly on script overview page, migrated script edit page, lesson edit page and teacher-facing lesson plan

I could not think of a good way to add an end-to-end test for the steps covered by manual testing above. the reason is because we can't afford to make filesystem changes in the test environment without risking interfering with other tests.

# Reviewer Checklist:
